### ISSUE TITLE
TFLearn Estimator Summary Writer Fails.

### ISSUE THREAD

**Issue by thesuperzapper** [tensorflow/tensorflow#7554]
### Issue:
TFLearn estimator summary writer fails because _write_dict_to_summary() in [/tensorflow/contrib/learn/python/learn/estimators/estimator.py](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/contrib/learn/python/learn/estimators/estimator.py#L324) dose not allow int values, which naturally stops the np.int64 valued 'global_step' parameter from being written, thus causing the summary writer to fail.

### How to reproduce:
Try and run [wide_n_deep_tutorial.py](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/examples/learn/wide_n_deep_tutorial.py), and you will get:

`INFO:tensorflow:Saving dict for global step 202: accuracy = 0.818254, accuracy/baseline_label_mean = 0.236226, accuracy/threshold_0.500000_mean = 0.818254, auc = 0.742517, global_step = 202, labels/actual_label_mean = 0.236226, labels/prediction_mean = 0.182356, loss = 0.680463, precision/positive_threshold_0.500000_mean = 0.764145, recall/positive_threshold_0.500000_mean = 0.333593
WARNING:tensorflow:Skipping summary for global_step, must be a float or np.float32.`

### Pull Request:
#7555

**Comment by aselle** [comment]
@ilblackdragon , @martinwicke, could you please take a look at this issue! Thanks!

**Comment by martinwicke** [comment]
Just looking over this again, why does this fail? It emits a warning and does not write the global_step, which is probably not a bad thing. Is there an actual failure?

**Comment by facundesg** [comment]
Hello, I'm facing the same warning as @thesuperzapper that prevents me from writing to the summary and visualizing on tensorboard.

**Comment by rjpg** [comment]
same problem using early stopping ... no output to see the values (accuracy).. and it seams the early stopping does not "activate" and stops as it should.

I have this monitor to test every step (just for debug): 

`validation_monitor = tf.contrib.learn.monitors.ValidationMonitor(
    x=test_inputs,
    y=test_output,
    eval_steps=1,  # Try adding this
    every_n_steps=1)`

and this output :
## ...
`Please switch to tf.summary.scalar. Note that tf.summary.scalar uses the node name instead of the tag. This means that TensorFlow will automatically de-duplicate summary names based on the scope they are created in. Also, passing a tensor or list of tags to a scalar summary op is no longer supported.
INFO:tensorflow:Starting evaluation at 2017-04-27-16:05:14
INFO:tensorflow:Restoring parameters from C:\Users\birinhos\AppData\Local\Temp\tmpn8b3fjgp\model.ckpt-1
INFO:tensorflow:Evaluation [1/1]
INFO:tensorflow:Finished evaluation at 2017-04-27-16:05:15
INFO:tensorflow:Saving dict for global step 1: accuracy = 0.221193, global_step = 1, loss = 1.6383
WARNING:tensorflow:Skipping summary for global_step, must be a float or np.float32.
INFO:tensorflow:Validation (step 1): loss = 1.6383, global_step = 1, accuracy = 0.221193
INFO:tensorflow:global_step/sec: 5.17624
INFO:tensorflow:loss = 1.59089, step = 101 (19.303 sec)
INFO:tensorflow:global_step/sec: 6.90738
INFO:tensorflow:loss = 1.58918, step = 201 (14.477 sec)`
 
##...

`INFO:tensorflow:loss = 1.52565, step = 4001 (13.839 sec)
INFO:tensorflow:global_step/sec: 6.71895
INFO:tensorflow:loss = 1.52449, step = 4101 (14.881 sec)
INFO:tensorflow:global_step/sec: 7.22961
INFO:tensorflow:loss = 1.52349, step = 4201 (13.835 sec)
INFO:tensorflow:Saving checkpoints for 4265 into C:\Users\birinhos\AppData\Local\Temp\tmpn8b3fjgp\model.ckpt.
WARNING:tensorflow:From C:\Users\birinhos\AppData\Local\Programs\Python\Python35\lib\site-packages\tensorflow\contrib\learn\python\learn\monitors.py:662: calling BaseEstimator.evaluate (from tensorflow.contrib.learn.python.learn.estimators.estimator) with y is deprecated and will be removed after 2016-12-01.
Instructions for updating:
Estimator is decoupled from Scikit Learn interface by moving into
separate class SKCompat. Arguments x, y and batch_size are only
available in the SKCompat class, Estimator will only accept input_fn.
Example conversion:
  est = Estimator(...) -> est = SKCompat(Estimator(...))
WARNING:tensorflow:From C:\Users\birinhos\AppData\Local\Programs\Python\Python35\lib\site-packages\tensorflow\contrib\learn\python\learn\monitors.py:662: calling BaseEstimator.evaluate (from tensorflow.contrib.learn.python.learn.estimators.estimator) with x is deprecated and will be removed after 2016-12-01.
Instructions for updating:
Estimator is decoupled from Scikit Learn interface by moving into
separate class SKCompat. Arguments x, y and batch_size are only
available in the SKCompat class, Estimator will only accept input_fn.
Example conversion:
  est = Estimator(...) -> est = SKCompat(Estimator(...))
WARNING:tensorflow:From C:\Users\birinhos\AppData\Local\Programs\Python\Python35\lib\site-packages\tensorflow\contrib\learn\python\learn\estimators\head.py:615: scalar_summary (from tensorflow.python.ops.logging_ops) is deprecated and will be removed after 2016-11-30.
Instructions for updating:
Please switch to tf.summary.scalar. Note that tf.summary.scalar uses the node name instead of the tag. This means that TensorFlow will automatically de-duplicate summary names based on the scope they are created in. Also, passing a tensor or list of tags to a scalar summary op is no longer supported.
INFO:tensorflow:Starting evaluation at 2017-04-27-16:15:14
INFO:tensorflow:Restoring parameters from C:\Users\birinhos\AppData\Local\Temp\tmpn8b3fjgp\model.ckpt-4265
INFO:tensorflow:Evaluation [1/1]
INFO:tensorflow:Finished evaluation at 2017-04-27-16:15:16
INFO:tensorflow:Saving dict for global step 4265: accuracy = 0.290535, global_step = 4265, loss = 1.55965
WARNING:tensorflow:Skipping summary for global_step, must be a float or np.float32.
INFO:tensorflow:Validation (step 4265): loss = 1.55965, global_step = 4265, accuracy = 0.290535
INFO:tensorflow:global_step/sec: 5.2094
INFO:tensorflow:loss = 1.52247, step = 4301 (19.196 sec)
INFO:tensorflow:global_step/sec: 7.4686
INFO:tensorflow:loss = 1.52147, step = 4401 (13.443 sec)`


##...

**Comment by roopsmall** [comment]
I'm also getting this warning using `global_step=tf.contrib.framework.get_global_step()`. Using tensorflow v1.1.0

**Comment by rjpg** [comment]
hello 

I have solve this problem :

adding config to the DNNClassifier : 
    config=tf.contrib.learn.RunConfig(save_checkpoints_steps=save_checkpoints_steps
                                                                    ,save_checkpoints_secs=None)

somehow it forces the call of the monitor every  "save_checkpoints_steps"  you can configure the "force call" of the monitor by seconds. I prefer by steps that is why this parameter is set to None:  save_checkpoints_secs=None....
